irohad/ametsuchi/impl/rocksdb_common.hpp

```diff
@@ -0,0 +1,1203 @@
+                     std::string_view account,
+                     std::string_view domain) {
+    assert(!domain.empty());
+    assert(!account.empty());
+
+    /// TODO(iceseer): remove this vector!
+    std::vector<std::string> roles;
+    auto status = enumerateKeys(common,
+                                [&](auto role) {
+                                  if (!role.empty())
+                                    roles.emplace_back(role.ToStringView());
+                                  else {
+                                    assert(!"Role can not be empty string!");
+                                  }
+                                  return true;
+                                },
+                                fmtstrings::kPathAccountRoles,
+                                domain,
+                                account);
+
+    if (!status.ok())
+      return makeError<shared_model::interface::RolePermissionSet>(
+          3,
+          "Enumerate account {}#{} roles failed with status: {}.",
+          account,
+          domain,
+          status.ToString());
+
+    if (roles.empty())
+      return makeError<shared_model::interface::RolePermissionSet>(
+          4, "Account {}#{} have ho roles.", account, domain);
+
+    shared_model::interface::RolePermissionSet permissions;
+    for (auto &role : roles) {
+      auto opt_perm =
+          forRole<kDbOperation::kGet, kDbEntry::kMustExist>(common, role);
+      RDB_ERROR_CHECK(opt_perm);
+      permissions |= *opt_perm.assumeValue();
+    }
+
+    return permissions;
+  }
+
+  inline expected::Result<void, DbError> checkPermissions(
+      shared_model::interface::RolePermissionSet const &permissions,
+      shared_model::interface::permissions::Role const to_check) {
+    if (permissions.isSet(to_check))
+      return {};
+
+    return makeError<void>(11, "No permissions.");
+  }
+
+  inline expected::Result<void, DbError> checkPermissions(
+      std::string_view domain_id,
+      std::string_view creator_domain_id,
+      shared_model::interface::RolePermissionSet const &permissions,
+      shared_model::interface::permissions::Role const all,
+      shared_model::interface::permissions::Role const domain) {
+    if (permissions.isSet(all))
+      return {};
+
+    if (domain_id == creator_domain_id && permissions.isSet(domain))
+      return {};
+
+    return makeError<void>(7, "No permissions.");
+  }
+
+  inline expected::Result<void, DbError> checkPermissions(
+      shared_model::interface::RolePermissionSet const &permissions,
+      shared_model::interface::GrantablePermissionSet const
+          &grantable_permissions,
+      shared_model::interface::permissions::Role const role,
+      shared_model::interface::permissions::Grantable const granted) {
+    if (permissions.isSet(role))
+      return {};
+
+    if (grantable_permissions.isSet(granted))
+      return {};
+
+    return makeError<void>(14, "No permissions.");
```

Comment:
Maybe list all the error codes as named constants? Then users can check db error for something like `code == kNoPermissions`. Also these error codes do not correspond to codes used in the commands, so global named constants seem alright.